src/cpp/session/modules/SessionRUtil.R

```diff
@@ -91,12 +91,16 @@
 #' @param libPaths The library paths to be set and used by the child process.
 #'   By default, the parent's library paths are used.
 #'
+#' @param returnVal Whether or not the callback should return a value. By
+#'   default, nothing will be returned.
+#'
 #' @param ... Optional arguments passed to `system2()`.
 #' 
 .rs.addFunction("executeFunctionInChildProcess", function(callback, 
                                                           data = list(),
                                                           workingDir = NULL,
                                                           libPaths = .libPaths(),
+                                                          returnVal = FALSE,
```

Comment:
I think I would prefer leaving this parameter out, and just always allowing for a return value. The only extra piece we might want to document is that, since we're using RDS for serialization, we should make sure any functions run through this interface return a value that is relatively cheap to serialize.